overlord/snapstate/snapstate.go

```diff
@@ -462,20 +462,66 @@ func UpdateMany(st *state.State, names []string, userID int) ([]string, []*state
 		}
 	}
 
-	updated := make([]string, 0, len(updates))
+	params := func(update *snap.Info) (string, Flags, *SnapState) {
```

Comment:
Mostly curious why you pass the func here instead of the nacked map[string]*SnapState ? Its nice, but I wonder if there is a deeper reason? (and not important to answer today, its already late :)